src/pubsub/ua_pubsub_reader.c

```diff
@@ -291,6 +308,13 @@ UA_Server_addReaderGroup(UA_Server *server, UA_NodeId connectionIdentifier,
         return UA_STATUSCODE_BADCONFIGURATIONERROR;
     }
 
+    /* Regist (bind) the connection channel if it is not already registered */
```

Comment:
I a missing the same check for the WriterGroup.
Shouldn't that be handled in a symmetric way?